File: 28. Find the Index of the First Occurrence in a String/Solution-for-28.php
<?php

class Solution {

    /**
     * @param String $haystack
     * @param String $needle
     * @return Integer
     */
    function strStr($haystack, $needle) {
        // If needle is an empty string, return 0
        if ($needle === "") {
            return 0;
        }
        
        // Find the position of needle in haystack
        $pos = strpos($haystack, $needle);
        
        // If found, return the position; otherwise, return -1
        return ($pos !== false) ? $pos : -1;
    }
}

// // Example usage
// $solution = new Solution();

// // Output the result
// echo $solution->strStr("sadbutsad", "sad"); // 0
// echo $solution->strStr("leetcode", "leeto"); // -1

?>

File: 28. Find the Index of the First Occurrence in a String/Solution-for-28.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        # If needle is an empty string, return 0
        if needle == "":
            return 0
        
        # Find the position of needle in haystack
        pos = haystack.find(needle)
        
        # If found, return the position; otherwise, return -1
        return pos if pos != -1 else -1

# # Example usage
# solution = Solution()

# # Output the result
# print(f"{solution.strStr("sadbutsad", "sad")}") # 0
# print(f"{solution.strStr("leetcode", "leeto")}") # -1
